backend/src/main/java/ba/unsa/etf/si/bbqms/admin_service/implementation/DefaultAdminService.java

```diff
@@ -1,11 +1,106 @@
 package ba.unsa.etf.si.bbqms.admin_service.implementation;
 
 import ba.unsa.etf.si.bbqms.admin_service.api.AdminService;
+import ba.unsa.etf.si.bbqms.auth_service.api.RoleService;
+import ba.unsa.etf.si.bbqms.auth_service.api.UserService;
+import ba.unsa.etf.si.bbqms.domain.Role;
+import ba.unsa.etf.si.bbqms.domain.RoleName;
+import ba.unsa.etf.si.bbqms.domain.Tenant;
+import ba.unsa.etf.si.bbqms.domain.User;
+import ba.unsa.etf.si.bbqms.exceptions.AuthException;
+import ba.unsa.etf.si.bbqms.repository.UserRepository;
+import ba.unsa.etf.si.bbqms.tenant_service.api.TenantService;
+import ba.unsa.etf.si.bbqms.tfa_service.api.TwoFactorService;
+import ba.unsa.etf.si.bbqms.utils.UserValidator;
+import ba.unsa.etf.si.bbqms.ws.models.UserDto;
+import org.springframework.security.crypto.password.PasswordEncoder;
 import org.springframework.stereotype.Service;
 
+import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.Optional;
+import java.util.Set;
+
 @Service
 public class DefaultAdminService implements AdminService {
-    public String dummyAdminServiceImpl() {
-        return "Hello admin";
+
+    private final UserRepository userRepository;
+    private final UserService userService;
+    private final TwoFactorService twoFactorService;
+    private final TenantService tenantService;
+    private final RoleService roleService;
+    private final PasswordEncoder passwordEncoder;
+    public DefaultAdminService(final UserRepository userRepository,
+                               final UserService userService,
+                               final TwoFactorService twoFactorService,
+                               final TenantService tenantService,
+                               final RoleService roleService,
+                               final PasswordEncoder passwordEncoder) {
+        this.userRepository = userRepository;
+        this.userService = userService;
+        this.twoFactorService = twoFactorService;
+        this.tenantService = tenantService;
+        this.roleService = roleService;
+        this.passwordEncoder = passwordEncoder;
+    }
+
+    @Override
+    public List<User> findAdminsByCode(final String tenantCode){
+        final Set<RoleName> roleNameSet = Set.of(RoleName.ROLE_BRANCH_ADMIN);
```

Comment:
What if we want an admin that isnt `ROLE_BRANCH_ADMIN`? We don't want to write a different method just for that. Method should be adjusted to allow passing role as parameter.